### PR TITLE
[ci] Ensure macOS builds use clang instead of gcc

### PR DESCRIPTION
**Pull request by mlin**
Appears to fix current main CI...unclear if setting gcc was intentional, and we do NOT do that for the wheel builds.

### CODE REVIEW

**Review by Shelnutt2 — APPROVED** [review]

File: .github/workflows/python-ci-full.yml
Comment (johnkerl) [inline .github/workflows/python-ci-full.yml:22]: > Appears to fix current main CI...unclear if setting gcc was intentional, and we do NOT do that for the wheel builds.

@mlin we can merge this & that's fine but the pre-merge minimal CI doesn't run on MacOS.

Just to verify this works, please _temporarily_ include a `- runs-on: macos-12` line in `python-ci-minimal.yml` and push & make sure it builds OK, then remove it, then merge to `main` with @Shelnutt2 's accpt

Comment (johnkerl) [reply .github/workflows/python-ci-full.yml:22]: @mlin I would do Request Changes but I want to be respectful of timezones. Please make this change as requested before merging. Thanks!

Comment (johnkerl) [reply .github/workflows/python-ci-full.yml:22]: I'm trying this advice on #1128 ... my own advice may have been misspelt ... 👀

**Review by johnkerl — CHANGES_REQUESTED** [review]
I tried the thing I suggested -- which is #1128 

This makes the change run on MacOS _before_ merge to `main` (not after) -- and it fails as at
https://github.com/single-cell-data/TileDB-SOMA/actions/runs/4432210297/jobs/7775996102

Maybe we need to explicitly specify `clang` -- ?

cc @Shelnutt2

**Comment by mlin** [comment]
Well spotted @johnkerl -- I didn't appreciate the minimal/full subtleties. Setting clang explicitly in the matrix [appears to work](https://github.com/single-cell-data/TileDB-SOMA/actions/runs/4433164052/jobs/7777942295). I added some breadcrumbs in the workflow .yml files for the next guy/gal. Please merge this if you see fit & I'm not online. cc @nguyenv @Shelnutt2

**Review by nguyenv — APPROVED** [review]

**Review by johnkerl — APPROVED** [review]

**Pull request merged**
